### PR TITLE
Made Flagger use downcasting

### PR DESCRIPTION
**Pull request by Verdagon**
Added a utility function for getting a nothing flag and a string flag.

Also fixed a couple bugs.

I'm experimenting with doc style, what do you think? It's a different tradeoff, don't know what I think about it yet.

### CODE REVIEW

File: src/flagger/flagger.vale [hunk 1]
```diff
@@ -193,50 +193,52 @@ fn parse_all_flags<N>(flag_list [N * Flag], args Array<imm, str>) ParsedFlagList
...
 
         if (valid and flag_type != FLAG_NOTHING()) {
-            parsed_flag_list.parsed_flags!.add(parse_flag(args, flag_type, &i));
+            (new_parsed_flag, new_i) = parse_flag(args, flag_type, &i);
+            set i = new_i;
+            parsed_flag_list.parsed_flags!.add(new_parsed_flag);
         }
 
-        else if (flag_type == FLAG_NOTHING()) {
+        else if (valid and flag_type == FLAG_NOTHING()) {
```
Comment (Verdagon) [inline src/flagger/flagger.vale:209]: Not sure why adding this made things work lol

**Review by Verdagon — COMMENTED** [review]
LMK if i should change anything!

**Comment by TheMoxyFoxy** [comment]
Will review within the next few days

**Comment by TheMoxyFoxy** [comment]
Looked over the commit when I had a moment. Looks great. Will review it in action once I have the chance

**Pull request merged**